Reserve `sarif` crate

TimDiekmann opened the pull request:
## 🌟 What is the purpose of this PR?

As discussed offline, this reserves the `sarif` name on crates.io.

## 🔍 What does this change?

- Adds boilerplate to define a crate in HASH

## 🛡 What tests cover this?

- /s The crate currently has 💯% line coverage _and_ branch coverage

## ❓ How to test this?

- Verify the links in the README and potential typos as this will be published.
- /s Run the full test suite, this may take up to 0 seconds

indietyp commented on libs/sarif/Cargo.toml:
```diff
@@ -0,0 +1,12 @@
+readme = "README.md"
+repository = "https://github.com/hashintel/hash/tree/main/libs/sarif"
+keywords = ["sarif", "codeql", "serde", "static-analysis", "tools"]
+categories = ["data-structures", "development-tools", "parser-implementations", "parsing"]
```
is `parser-implementations` fitting here? AFAIK we do not plan to parse anything, just output (and transform) into valid SARIF.

TimDiekmann replied:
Probably leaving that out is good
```suggestion
categories = ["data-structures", "development-tools", "parsing"]
```

indietyp commented on libs/sarif/README.md:
```diff
@@ -0,0 +1,31 @@
+
+## Contributors
+
+`sarif` was created by [Tim Diekmann](https://github.com/TimDiekmann) and [Bilal Mahmoud](https://github.com/indietyp) to be used to improve DX around creating Pull Requests. It is being developed in conjunction with [HASH](https://hash.dev/) as an open-source project. We gratefully accept external contributions and have published a [contributing guide](https://github.com/hashintel/hash/blob/main ...
```
❤️

indietyp reviewed (changes requested):
A couple of questions; otherwise looks good. This also needs:

* CODEOWNERS (@nonparibus)
* renovate needs to be configured.

TimDiekmann commented:
Thanks, @indietyp, I added the labeller and renovate configuration. As a drive-by I also added `deer` to the publish patterns (we forgot to add it) and grouped the Rust toolchain bumps in renovate to avoid 5 PRs per week.
